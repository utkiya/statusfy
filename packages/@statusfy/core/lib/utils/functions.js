const { fse, grayMatter, chalk, path } = require("@statusfy/common");

const getIncidentsFromProject = async contentDir => {
  const files = await fse.readdir(contentDir);
  const incidentsList = [];

  for (let i = 0; i < files.length; i++) {
    const f = path.resolve(contentDir, files[i]);
    const ext = path.extname(f);
    const fileName = path.basename(f);

    if (ext === ".md") {
      const fileContent = await fse.readFile(f);
      const { data } = grayMatter.parse(fileContent);

      incidentsList.push({
        value: {
          name: fileName,
          path: f
        },
        name: `${fileName} > ${chalk.yellow(data.title)} (${chalk.green(
          new Date(data.date).toUTCString()
        )})`
      });
    }
  }

  return incidentsList;
};

const generateIncident = (data, content, format) => {
  let matterContent = grayMatter.stringify(content, data, format);

  if (["json", "toml"].includes(format)) {
    matterContent = matterContent.replace(/^---/, `---${format}`);
  }

  return matterContent;
};

module.exports = {
  getIncidentsFromProject,
  generateIncident
};
