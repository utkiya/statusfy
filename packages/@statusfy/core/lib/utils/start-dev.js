const portfinder = require("portfinder");
const bs = require("browser-sync").create();

const { logger, path } = require("@statusfy/common");
const createServer = require("../../server");
const generateConfig = require("../config/generate");

/* eslint-disable require-await */
const start = async () => {
  process.env.NODE_ENV = "development";
  const sourceDir = process.env.STATUSFY_SOURCE_DIR;
  const cliOptions = JSON.parse(process.env.STATUSFY_CLI_OPTIONS);

  // Generate configuration
  const { nuxtConfig, siteConfig } = generateConfig(sourceDir, cliOptions);
  nuxtConfig.mode = cliOptions.ssr ? "universal" : "spa";
  delete nuxtConfig.build.analyze;

  portfinder
    .getPortPromise()
    .then(async port => {
      const serverPort = nuxtConfig.server.port;

      // Update server port
      nuxtConfig.server.port = port;

      // Initialize the server
      await createServer(
        siteConfig,
        nuxtConfig,
        nuxtConfig.server.host,
        nuxtConfig.server.port
      );

      // Create borwser-sync server
      bs.init({
        port: serverPort,
        proxy: `${nuxtConfig.server.host}:${nuxtConfig.server.port}`,
        open: false
      });

      bs.watch(path.join(sourceDir, siteConfig.content.dir, "**/*.md")).on(
        "change",
        bs.reload
      );
    })
    .catch(err => {
      logger.error(err);
    });
};

start();
