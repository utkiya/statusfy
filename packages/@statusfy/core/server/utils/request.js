module.exports = req => {
  const language = req.params.lang || req.app.get("language");
  const page = req.query.page || req.params.lang || 1;

  return {
    language,
    page,
    siteConfig: req.app.get("siteConfig")
  };
};
