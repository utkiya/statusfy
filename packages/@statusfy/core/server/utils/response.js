module.exports = (res, lang) => ({
  json(data) {
    res.setHeader("Content-Language", lang);
    res.json(data);
  },
  notFound(message = "Not Found") {
    res.setHeader("Cache-Control", "no-cache");
    res.status(404).json({
      message
    });
  }
});
