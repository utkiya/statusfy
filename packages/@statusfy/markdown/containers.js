const container = require('markdown-it-container')
const { validator } = require('@statusfy/common')

module.exports = md => {
  md.use(...createUpdateContainer('update'))
}

function createUpdateContainer (klass) {
  const parseInfo = (info) => info.match(/^update\s+(.*)\s+\|\s+(.*)$/)

  return [container, klass, {
    validate: function (params) {
      const match = parseInfo(params.trim())

      if (match !== null) {
        // Make sure the date is in RFC3339 format
        return validator.isRFC3339(match[2])
      } else {
        return false
      }
    },
    render (tokens, idx) {
      const token = tokens[idx]

      if (token.nesting === 1) {
        const match = parseInfo(token.info.trim())
        const title = match[1].trim()
        const date = new Date(match[2]).toISOString()

        return `<div class="update-block">
          <div class="update-block-date">${date}</div>
          <div class="update-block-content">
            <div class="update-block-title">${title}</div>
            <div class="update-block-body">\n`
      } else {
        return `</div></div></div>\n`
      }
    }
  }]
}
