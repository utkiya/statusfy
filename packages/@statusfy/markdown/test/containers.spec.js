import { md, getFragment } from './utils'
import containers from '@/markdown/containers.js'

const mdC = md().use(containers)

describe('containers', () => {
  const containerLabels = ['investigating', 'resolved', 'monitoring']

  containerLabels.forEach(label => {
    test(label, async () => {
      const input = await getFragment(`container-${label}`)
      const output = mdC.render(input)

      expect(output).toMatchSnapshot()
    })
  })
})
