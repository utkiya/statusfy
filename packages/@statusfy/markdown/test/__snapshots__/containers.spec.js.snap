// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers investigating 1`] = `
<div class="update-block">
  <div class="update-block-date">2018-04-24T06:39:09.000Z</div>
  <div class="update-block-content">
    <div class="update-block-title">Investigating</div>
    <div class="update-block-body">
      <p>Our <strong>Files Conversion</strong> Process is responding slowly and we've investigating what is causing this issue.</p>
    </div>
  </div>
</div>
`;

exports[`containers monitoring 1`] = `
<div class="update-block">
  <div class="update-block-date">2018-04-12T03:45:12.000Z</div>
  <div class="update-block-content">
    <div class="update-block-title">Monitoring</div>
    <div class="update-block-body">
      <p>We've added capacity to our origin and are seeing improved system behavior. We continue to monitor the situation.</p>
    </div>
  </div>
</div>
`;

exports[`containers resolved 1`] = `
<div class="update-block">
  <div class="update-block-date">2018-04-24T07:24:09.000Z</div>
  <div class="update-block-content">
    <div class="update-block-title">Resolved</div>
    <div class="update-block-body">
      <p>The Conversion Process is back to normal and we'll continue to monitor.</p>
      <p>:tada: :100:</p>
    </div>
  </div>
</div>
`;
